lib/ex_doc/formatter/epub/templates.ex

```diff
@@ -68,4 +68,18 @@ defmodule ExDoc.Formatter.EPUB.Templates do
   EEx.function_from_file(:defp, :head_template,
                          Path.expand("templates/head_template.eex", __DIR__),
                          [:config, :page])
+
+  "templates/media-types.txt"
+  |> Path.expand(__DIR__)
+  |> File.read!()
+  |> String.split("\n", trim: true)
+  |> Enum.each(fn(line) ->
+    [extension, media] = String.split(line, ",")
+
+    def media_type("." <> unquote(extension)) do
+      unquote(media)
+    end
+  end)
+
+  def media_type(_), do: raise "publication resource not supported in EPUB format"
```

Comment:
Let's include the argument in the error message and try to make it clearer. Something like:

    raise "asset with extension #{inspect arg} is not supported by EPUB format"

What do you think?